### PR TITLE
[INLONG-5628][SortStandalone] Add buffer limit of sink dispatch queue

### LINKED ISSUE #5628: [Feature][SortStandalone] Add buffer limit for dispatch queue of the sink
### Description

Add buffer limit of sink dispatch queue.
Avoid that the dispatch queue is too big and JVM go to Full GC.

### Use case

_No response_

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by luchunliang**
### Prepare a Pull Request
- Title: [INLONG-5628][SortStandalone] Add buffer limit of sink dispatch queue
- Fixes #5628 

### Motivation

*Explain here the context, and why you're making that change. What is the problem you're trying to solve?*

### Modifications

*Describe the modifications you've done.*

### Verifying this change

*(Please pick either of the following options)*

- [ ] This change is a trivial rework/code cleanup without any test coverage.

- [ ] This change is already covered by existing tests, such as:
  *(please describe tests)*

- [ ] This change added tests and can be verified as follows:

  *(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (10MB)*
  - *Extended integration test for recovery after broker failure*

### Documentation

  - Does this pull request introduce a new feature? (yes / no)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)
  - If a feature is not applicable for documentation, explain why?
  - If a feature is not documented yet in this PR, please create a follow-up issue for adding the documentation

### CODE REVIEW

File: inlong-dataproxy/dataproxy-source/src/main/java/org/apache/inlong/dataproxy/sink/SinkContext.java [hunk 1]
```diff
@@ -37,19 +40,22 @@ public class SinkContext {
...
+    public static final int DEFAULT_MAX_BUFFERQUEUE_SIZE_KB = 128 * 1024;
 
-    protected final String proxyClusterId;
+    protected final String clusterId;
     protected final String sinkName;
     protected final Context sinkContext;
 
     protected final Channel channel;
-
+    //
```
Comment (vernedeng) [inline inlong-dataproxy/dataproxy-source/src/main/java/org/apache/inlong/dataproxy/sink/SinkContext.java:54]: useless code

File: inlong-dataproxy/dataproxy-source/src/main/java/org/apache/inlong/dataproxy/sink/SinkContext.java [hunk 2]
```diff
@@ -37,19 +40,22 @@ public class SinkContext {
...
     protected final int maxThreads;
     protected final long processInterval;
     protected final long reloadInterval;
-
+    //
```
Comment (vernedeng) [inline inlong-dataproxy/dataproxy-source/src/main/java/org/apache/inlong/dataproxy/sink/SinkContext.java:58]: ditto

**Review by vernedeng — APPROVED** [review]
lgtm

File: inlong-dataproxy/dataproxy-source/src/main/java/org/apache/inlong/dataproxy/sink/SinkContext.java (hunk 1 above)
Comment (luchunliang) [reply inlong-dataproxy/dataproxy-source/src/main/java/org/apache/inlong/dataproxy/sink/SinkContext.java]: remove it

File: inlong-dataproxy/dataproxy-source/src/main/java/org/apache/inlong/dataproxy/sink/SinkContext.java (hunk 2 above)
Comment (luchunliang) [reply inlong-dataproxy/dataproxy-source/src/main/java/org/apache/inlong/dataproxy/sink/SinkContext.java]: remove it

File: inlong-dataproxy/dataproxy-source/src/main/java/org/apache/inlong/dataproxy/sink/DefaultEventHandler.java [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static org.apache.inlong.sdk.commons.protocol.EventConstants.HEADER_CACHE_VERSION_1;
+import static org.apache.inlong.sdk.commons.protocol.EventConstants.HEADER_KEY_VERSION;
+
+/**
+ * DefaultEventHandler
+ *
```
Comment (healchow) [inline inlong-dataproxy/dataproxy-source/src/main/java/org/apache/inlong/dataproxy/sink/DefaultEventHandler.java:43]: Unnecessary blank lines.

File: inlong-dataproxy/dataproxy-source/src/main/java/org/apache/inlong/dataproxy/sink/EventHandler.java [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+
+import org.apache.inlong.dataproxy.config.pojo.IdTopicConfig;
+import org.apache.inlong.dataproxy.dispatch.DispatchProfile;
+import org.apache.inlong.sdk.commons.protocol.ProxySdk.INLONG_COMPRESSED_TYPE;
+
+import java.util.Map;
+
+/**
+ * IEventHandler
+ *
```
Comment (healchow) [inline inlong-dataproxy/dataproxy-source/src/main/java/org/apache/inlong/dataproxy/sink/EventHandler.java:28]: Same as above.

File: inlong-dataproxy/dataproxy-source/src/main/java/org/apache/inlong/dataproxy/sink/SinkContext.java [hunk 5]
```diff
@@ -114,16 +120,16 @@ public void reload() {
 
     /**
      * get clusterId
-     *
+     *
```
Comment (healchow) [inline inlong-dataproxy/dataproxy-source/src/main/java/org/apache/inlong/dataproxy/sink/SinkContext.java:123]: Please do not add a space at the end of the line.

File: inlong-dataproxy/dataproxy-source/src/main/java/org/apache/inlong/dataproxy/sink/kafkazone/KafkaZoneProducer.java [hunk 6]
```diff
@@ -64,6 +64,7 @@ public KafkaZoneProducer(String workerName, KafkaZoneSinkContext context) {
      */
     public void start() {
         try {
+            LOG.info("start KafkaZoneProducer:{}", workerName);
```
Comment (healchow) [inline inlong-dataproxy/dataproxy-source/src/main/java/org/apache/inlong/dataproxy/sink/kafkazone/KafkaZoneProducer.java:67]: Suggested change to the end of `this.setReloadTimer();`, and change it to:
```
LOG.info("start KafkaZoneProducer success for {}", workerName);
```

File: inlong-dataproxy/dataproxy-source/src/main/java/org/apache/inlong/dataproxy/sink/SinkContext.java (hunk 5 above)
Comment (luchunliang) [reply inlong-dataproxy/dataproxy-source/src/main/java/org/apache/inlong/dataproxy/sink/SinkContext.java]: fixed

File: inlong-dataproxy/dataproxy-source/src/main/java/org/apache/inlong/dataproxy/sink/DefaultEventHandler.java (hunk 3 above)
Comment (luchunliang) [reply inlong-dataproxy/dataproxy-source/src/main/java/org/apache/inlong/dataproxy/sink/DefaultEventHandler.java]: fixed

File: inlong-dataproxy/dataproxy-source/src/main/java/org/apache/inlong/dataproxy/sink/kafkazone/KafkaZoneProducer.java (hunk 6 above)
Comment (luchunliang) [reply inlong-dataproxy/dataproxy-source/src/main/java/org/apache/inlong/dataproxy/sink/kafkazone/KafkaZoneProducer.java:67]: fixed

File: inlong-dataproxy/dataproxy-source/src/main/java/org/apache/inlong/dataproxy/sink/EventHandler.java (hunk 4 above)
Comment (luchunliang) [reply inlong-dataproxy/dataproxy-source/src/main/java/org/apache/inlong/dataproxy/sink/EventHandler.java]: fixed

**Review by gosonzhang — APPROVED** [review]

**Comment by gosonzhang** [comment]
@luchunliang, the PR has conflicts that must be resolved, and I see that LinkedBlockingQueue is used in some sinks, but there is no corresponding replacement for BufferQueue. Can it be replaced uniformly?

**Review by gosonzhang — CHANGES_REQUESTED** [review]
I see that LinkedBlockingQueue is used in some sinks, but there is no corresponding replacement for BufferQueue. Can it be replaced uniformly?

**Pull request closed without merging**